src/System.CommandLine.Tests/ParserTests.cs

```diff
@@ -402,6 +402,69 @@ public void Optional_option_arguments_are_unbundled()
                   .BeEquivalentTo("-a", "-b", "-c");
         }
 
+        [Theory]
+        [InlineData("-abcvalue", null, "value")]
+        [InlineData("-abc value", null, "value")]
+        [InlineData("-abc=value", null, "value")]
+        [InlineData("-abc:value", null, "value")]
+        [InlineData("-abvalue", "value", null)]
+        [InlineData("-ab value", "value", null)]
+        [InlineData("-ab=value", "value", null)]
+        [InlineData("-ab:value", "value", null)]
+        [InlineData("-abvalue -cvalue", "value", "value")]
+        // a should always be set
+        // b should always be set but value is optional
+        // c is optional
+        public void Last_bundled_option_can_accept_arguments(string arguments, string bvalue, string cvalue)
+        {
+            var optionA = new Option("-a");
+            var optionB = new Option("-b")
+            {
+                Argument = new Argument<string>
+                {
+                    Arity = ArgumentArity.ZeroOrOne
+                }
+            };
+            var optionC = new Option("-c") 
+            {
+                Argument = new Argument<string>
+                {
+                    Arity = ArgumentArity.ExactlyOne
+                }
+            };
+
+            var command = new RootCommand
+            {
+                optionA,
+                optionB,
+                optionC
+            };
+
+            var result = command.Parse(arguments);
+            result.HasOption(optionA).Should().BeTrue();
+            result.HasOption(optionB).Should().BeTrue();
+
+            if (bvalue != null)
```

Comment:
Can we break this up into multiple unit test so the tests don't need to be conditional?

Reply:
Will do. Just thought this was more DRY.

Reply:
I've unbundled the unbundling test 😛.